pkg/templatelookup/mandatory.go

```diff
@@ -4,23 +4,27 @@ import (
 	"context"
 	"fmt"
 
+	k8slabels "k8s.io/apimachinery/pkg/labels"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/kyma-project/lifecycle-manager/api/shared"
 	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
 )
 
 // GetMandatory returns ModuleTemplates TOs (Transfer Objects) which are marked are mandatory modules.
 func GetMandatory(ctx context.Context, kymaClient client.Reader) (ModuleTemplatesByModuleName,
 	error,
 ) {
-	moduleTemplateList := &v1beta2.ModuleTemplateList{}
-	if err := kymaClient.List(ctx, moduleTemplateList); err != nil {
-		return nil, fmt.Errorf("could not list ModuleTemplates: %w", err)
+	mandatoryModuleTemplateList := &v1beta2.ModuleTemplateList{}
+	labelSelector := k8slabels.SelectorFromSet(k8slabels.Set{shared.IsMandatoryModule: shared.EnableLabelValue})
+	if err := kymaClient.List(ctx, mandatoryModuleTemplateList,
+		&client.ListOptions{LabelSelector: labelSelector}); err != nil {
+		return nil, fmt.Errorf("could not list mandatory ModuleTemplates: %w", err)
```

Comment:
Maybe we could also add a field selector `client.MatchingFields{"metadata.deletionTimestamp": ""}`. Then we could get rid of filtering in the for loop bellow.

Reply:
Maybe this could work? 

```go
// GetMandatory returns ModuleTemplates TOs (Transfer Objects) which are marked are mandatory modules.
func GetMandatory(ctx context.Context, kymaClient client.Reader) (ModuleTemplatesByModuleName,
	error,
) {
	mandatoryModuleTemplateList := &v1beta2.ModuleTemplateList{}
	labelSelector := k8slabels.SelectorFromSet(k8slabels.Set{shared.IsMandatoryModule: shared.EnableLabelValue})
	fieldSeleector := fields.OneTermEqualSelector("metadata.deletionTimestamp", "")
	if err := kymaClient.List(ctx, mandatoryModuleTemplateList,
		&client.ListOptions{LabelSelector: labelSelector, FieldSelector: fieldSeleector}); err != nil {
		return nil, fmt.Errorf("could not list mandatory ModuleTemplates: %w", err)
	}

	mandatoryModules := make(map[string]*ModuleTemplateInfo)
	for _, moduleTemplate := range mandatoryModuleTemplateList.Items {
		mandatoryModules[moduleTemplate.Name] = &ModuleTemplateInfo{
			ModuleTemplate: &moduleTemplate,
			Err:            nil,
		}
	}
	return mandatoryModules, nil
}

```

Reply:
no no never mind I think I was doing some other thing that made it not working. I am trying it now with the correct value